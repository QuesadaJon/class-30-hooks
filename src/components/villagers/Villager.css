.Villager {
  
  img {
    width: 150px;
  }
}