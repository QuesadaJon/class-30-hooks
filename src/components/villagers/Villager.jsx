import React from 'react';
import PropTypes from 'prop-types';
import styles from './Villager.css';

const Villager = ({ quote, name, image }) => (
  <figure className={styles.Villager}>
    <p>{name}</p>
    <img src={image} alt={name} />
    <figcaption>
      <p>{quote}</p>
    </figcaption>
  </figure>
);

Villager.propTypes = {
  quote: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
};

export default Villager;
