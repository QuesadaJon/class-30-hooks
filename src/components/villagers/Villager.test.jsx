import React from 'react';
import { render } from '@testing-library/react';
import Villager from './Villager';

describe('Villager component', () => {
  it('matches a snapshot', () => {
    const { asFragment } = render(<Villager
      name="Gunther"
      image="http://image.com"
      quote="hello"
    />);

    expect(asFragment()).toMatchSnapshot();
  });
});
