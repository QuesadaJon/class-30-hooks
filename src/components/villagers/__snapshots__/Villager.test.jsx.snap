// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Villager component matches a snapshot 1`] = `
<DocumentFragment>
  <figure>
    <p>
      Gunther
    </p>
    <img
      alt="Gunther"
      src="http://image.com"
    />
    <figcaption>
      <p>
        hello
      </p>
    </figcaption>
  </figure>
</DocumentFragment>
`;
